Implement ember-paper

gschorkopf opened the pull request:
📱 🎉 

Working list of TODO items:

- [x] All commented out code removed
- [x] All quotes are single-quotes where appropriate
- [x] Add extra space to EOF diffs
- [x] Clean up footer button styles, footer padding
- [x] Perhaps the "play" button should go to `rounds.new`
- [ ] Add styling to `rounds` table and `turns` scorekeeping pages that look rough after removing bootstrap
- [ ] Fix / stylize the "New Round" form so it's back to functional (user paper-chips for users?)
- [ ] Fix the autocomplete for users/courses so it's back to functional

Commit:
Install ember-paper

Commit:
Generate packakge-lock.json for npm5 users

Commit:
updated yarn.lock

Commit:
add paper-link addon

Commit:
remove bootstrap and import ember-paper

Commit:
Refactor layout

Commit:
Extract nav styles

Commit:
materialize the rounds page

Commit:
materialize round list

Commit:
add some padding

Commit:
terrible commit. Autocompletes themed and working. Some more UI changes

Commit:
not sure what this...

cmcculler committed:
make rounds.index the 'home' screen

cmcculler committed:
move rounds.index redirect out of application and into a root index route

cmcculler committed:
linking to round detail

cmcculler committed:
remove jsconfig.json

cmcculler committed:
ignore editor specific files

cmcculler committed:
break out variables

cmcculler committed:
Rounds styles

cmcculler committed:
use variables for colors

cmcculler committed:
Cardify rounds list

gschorkopf committed:
Various cleanups: removes comments, adds EOF spaces, single quotes

gschorkopf committed:
Satiates the appetite of the ever-hungry ESLinter

gschorkopf commented:
See code comments here: https://github.com/frolfr/frolfr-web/pull/3

gschorkopf committed:
Adds paper-buttons / paper-chips to templates

gschorkopf committed:
Adds some basic transitions

gschorkopf committed:
Fixes footer buttons / padding

Not sure if this is the best way to resolve this. Previously, it looked
like there was a button within a button on the footer. This fixes that
style

gschorkopf committed:
Updates main button to route to rounds.new

gschorkopf committed:
Reintroduces pagination on the rounds page

And moves the tabs to the rounds.index (they shouldn’t exist on
rounds.new)

gschorkopf committed:
Defaults to an anonymous avatar picture

gschorkopf committed:
Cleans up some constants in models/user

cmcculler committed:
Implement friends route

cmcculler committed:
small tweak to round-resource

cmcculler committed:
styled implementation of material footer nav

cmcculler committed:
implement bottom-nav component to contain the bottom navbar

cmcculler committed:
implement bottom-nav-item component

cmcculler committed:
use the actual score

cmcculler committed:
add some small transitions to the bottom nav for a more material feel

cmcculler committed:
Begin reworking the new round page

gschorkopf committed:
Removes unused tests

Commit:
Select a course working

gschorkopf commented on app/components/round-form.js:
```diff
@@ -59,6 +86,20 @@ export default Ember.Component.extend({
+
+    closeFriendsDialog(event) {
+      if(event === 'save') {
+        // do ssomething
```
What's going on here?

ultimatemonty replied:
Placeholder to make sure things worked like I wanted. Will implement when the "who are you playing with" dialog is implemented

gschorkopf commented on app/transitions.js:
```diff
@@ -54,4 +54,15 @@ export default function(){
+
+  this.transition(
+  this.hasClass('liquid-dialog-container'),
+  this.use('explode', {
```
💥

gschorkopf commented on app/styles/app.scss:
```diff
@@ -1,5 +1,7 @@
 @import 'variables';
 @import 'ember-paper';
+@import "ember-paper-expansion-panel";
```
Where'd this come from? Are there other things we can import (like table styles)?

ultimatemonty replied:
Ugh I missed removing that. I toyed around with expansion panels in the select course dialog before sticking with tabs

ultimatemonty replied:
Table styles should be there by default. See http://miguelcobain.github.io/ember-paper/#/typography

gschorkopf replied:
👍 I was gonna say, I got `Error: File to import not found or unreadable: ember-paper-expansion-panel.`

Also... if the table styles should be there, I wonder why they're not?

Commit:
Add ember-array-contains-helper

Commit:
Make sure we're checking for unused props via linting

Commit:
Remove un-used import

Commit:
Selecting friends for round working

cmcculler committed:
setup ui service to help control some ui state (fab, bottom-nav visiblity)

cmcculler committed:
destructure :allthethings:

cmcculler committed:
remove package-lock.json. supporting two package managers sucks

cmcculler committed:
Add save and cancel buttons to new round form. Change some layout stuff, cuss alot at flexbox

cmcculler committed:
update disabled routes to include round.turn

gschorkopf commented on app/services/ui.js:
```diff
@@ -6,13 +6,13 @@ export default Service.extend({
 
   isFabVisible: computed('router.currentRouteName', function() {
-    const disabledRoutes = [ 'rounds.new', 'rounds.current' ];
+    const disabledRoutes = [ 'rounds.new', 'round.turns' ];
```
Neat!

gschorkopf commented on app/components/round-form.js:
```diff
@@ -4,8 +4,9 @@ const { Component, computed, inject } = Ember;
 
 export default Component.extend({
-  classNames: ['round-form'],
+  classNames: ['round-form', 'flex-100'],
```
What's `flex-100`?

gschorkopf commented:
Will rebase.

gschorkopf commented:
I don't know how valuable the `friends` route is going to be without being able to sign in. Maybe let's move friends to a separate branch?

gschorkopf commented:
Rebased delicately, based on the eye test looks like we're good.

gschorkopf committed:
Installs ember-simple-auth

gschorkopf committed:
Sets an after authentication route

gschorkopf committed:
Adds a basic login button / avatar to the toolbar

gschorkopf committed:
Adds authenticator/authorizor and passes to adapter

gschorkopf committed:
Adds ability to login, adds sessionAccount service to handle current user

gschorkopf committed:
Removes concept of "friends"

We should think about new/better ways to conceptualize friends. Right
now, I think being able to add any user in Frolfr to your round is fine

gschorkopf committed:
Fixes the cancel button

Cancel button was broken, router transition failed with “You passed the
`false` query parameter to transition”
No need to double-down on rolling back attributes, willDestroyElement()
handles that for us
This way there’s no need to inject the router

gschorkopf committed:
Removes unused style files

gschorkopf committed:
Initial pass at re-styling the scorecards table

gschorkopf committed:
Always center header tags

gschorkopf committed:
Refactors template logic into user-avatar component

This kinda messes up the round resource, but will update soon

gschorkopf committed:
Includes scorecards on rounds.index

* Saves a few API calls
